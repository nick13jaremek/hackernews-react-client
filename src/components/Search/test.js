import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Search from './index.js';

describe('Search', () => {
  const props = {
    value: 'test',
    onChange: () => true,
    onSubmit: () => false,
    children: 'Search',
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search { ...props } />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Search { ...props } />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

