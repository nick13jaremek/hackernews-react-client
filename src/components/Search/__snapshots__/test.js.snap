// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="test"
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
